dpctl/tensor/libtensor/include/kernels/sorting/topk.hpp

```diff
@@ -427,6 +443,7 @@ sycl::event topk_merge_impl(
                     [ctx, index_data] { sycl_free_noexcept(index_data, ctx); });
             });
 
+        index_data_owner.release();
```

Comment:
The same thing should be done here (move release to before and use a temporary for the pointer).